### ISSUE TITLE
population.txt file not found

### ISSUE THREAD

**Issue by ruairidhcumming** [konstantinstadler/pymrio#42]
I am trying to follow the Quickstart guide to exploring Exiobase2. when i call pymrio.parse_exiobase2 i get an error which i believe relates to the pymiro package itself missing a file. 
the path requested is in the pymiro package in my anaconda environment  C:\\Users\\CummingR\\AppData\\Local\\Continuum\\anaconda3\\envs\\RC37\\lib\\site-packages\\pymrio\\mrio_models\\exio20\\./misc/population.txt 
however the path only exists as far as 
C:\\Users\\CummingR\\AppData\\Local\\Continuum\\anaconda3\\envs\\RC37\\lib\\site-packages\\pymrio\\mrio_models
where the folder contains a __pycache__ folder, the test_miro folder and an __init__.py file. Is there a different file which needs to be installed manually? 
I have included the full error trace below.

exio2 = pymrio.parse_exiobase2(path=ExioFile,charact=True, popvector='exio2')
Traceback (most recent call last):

  File "<ipython-input-18-fb1c890f88e6>", line 1, in <module>
    exio2 = pymrio.parse_exiobase2(path=ExioFile,charact=True, popvector='exio2')

  File "C:\Users\CummingR\AppData\Local\Continuum\anaconda3\envs\RC37\lib\site-packages\pymrio\tools\ioparser.py", line 658, in parse_exiobase2
    index_col=0, sep='\t').astype(float)

  File "C:\Users\CummingR\AppData\Local\Continuum\anaconda3\envs\RC37\lib\site-packages\pandas\io\parsers.py", line 685, in parser_f
    return _read(filepath_or_buffer, kwds)

  File "C:\Users\CummingR\AppData\Local\Continuum\anaconda3\envs\RC37\lib\site-packages\pandas\io\parsers.py", line 457, in _read
    parser = TextFileReader(fp_or_buf, **kwds)

  File "C:\Users\CummingR\AppData\Local\Continuum\anaconda3\envs\RC37\lib\site-packages\pandas\io\parsers.py", line 895, in __init__
    self._make_engine(self.engine)

  File "C:\Users\CummingR\AppData\Local\Continuum\anaconda3\envs\RC37\lib\site-packages\pandas\io\parsers.py", line 1135, in _make_engine
    self._engine = CParserWrapper(self.f, **self.options)

  File "C:\Users\CummingR\AppData\Local\Continuum\anaconda3\envs\RC37\lib\site-packages\pandas\io\parsers.py", line 1917, in __init__
    self._reader = parsers.TextReader(src, **kwds)

  File "pandas\_libs\parsers.pyx", line 382, in pandas._libs.parsers.TextReader.__cinit__

  File "pandas\_libs\parsers.pyx", line 689, in pandas._libs.parsers.TextReader._setup_parser_source

FileNotFoundError: [Errno 2] File b'C:\\Users\\CummingR\\AppData\\Local\\Continuum\\anaconda3\\envs\\RC37\\lib\\site-packages\\pymrio\\mrio_models\\exio20\\./misc/population.txt' does not exist: b'C:\\Users\\CummingR\\AppData\\Local\\Continuum\\anaconda3\\envs\\RC37\\lib\\site-packages\\pymrio\\mrio_models\\exio20\\./misc/population.txt'

**Comment by GeorgeNash** [comment]
I am having the same issue.

`pymrio/mrio_models/exio20/./misc/population.txt' does not exist`

Any resolution?

**Comment by GeorgeNash** [comment]
I'm running on mac.

Reproducible as:

- Download ixi Exiobase V2: http://www.exiobase.eu/index.php/data-download/exiobase2-year-2007-full-data-set/78-mriot-ixi-fpa-coefficient-version2-2-2/file
- Decompress to a folder called `exiobase`

```
import pymrio
exio2 = pymrio.parse_exiobase2(path='exiobase', charact=True, popvector='exio2')
```

**Comment by konstantinstadler** [comment]
Hi,


I am currently on paternity leave - so there will be no new versions in the next couple of month. But I will fix that as soon as I am back.


Anyway, looking into the code, there might be a problem with the path joining for the population vector (line 656 in ioparser.py) - perhaps something that is OS specific. I have to check that when I am back (or fill free to send a merge request if you can fix it).


As a workaround: you can also pass a pandas dataframe to the "popvector" argument. So, you could download the population.txt file from

https://github.com/konstantinstadler/pymrio/tree/master/pymrio/mrio_models/exio20/misc

and save it somewhere convenient (/foo/bar/population.txt)


You can then do


import pandas as pd

import pymrio


pop_vec = pd.read_csv("/foo/bar/population.txt", sep="\t", index_col=0).as_type(float)

exio2 = pymrio.parse_exiobase2(path='exiobase', charact=True, popvector=pop_vec)


Might be that the read_csv is somehow different in the new pandas version (check the docs there).


Good luck

Konstantin




________________________________

**Comment by GeorgeNash** [comment]
Thanks Konstantin - much appreciated.

Enjoy your paternity leave.

**Comment by konstantinstadler** [comment]
This was solved in commit 6cadc3d (version 0.4.2d) and will be included in the next release
